packages/analytics-browser/src/config.ts

```diff
@@ -76,6 +78,7 @@ export class BrowserConfig extends Config implements IBrowserConfig {
     public transport: 'fetch' | 'xhr' | 'beacon' = 'fetch',
     public useBatch: boolean = false,
     userId?: string,
+    public diagnosticProvider: DiagnosticPlugin | undefined = new Diagnostic(),
```

Comment:
An error will return "Parameter cannot have question mark and initializer." if adding a question mark

What I want to achieve here is that 
- diagnosticProvider property can either hold a value of type DiagnosticPlugin or it can be undefined
   - users can opt out by passing undefined
   - the default value of this property is a Diagnostic instance

Reply:
[`new BrowserConfig()`](https://github.com/amplitude/Amplitude-TypeScript/blob/ebd668595b1f5adf1365a9d8106325c159d2119d/packages/analytics-browser/src/config.ts#L223-L224) passes an optional diagnosticProvider. So here must accept undefined as well. Then two options left:
1. `public diagnosticProvider?: DiagnosticPlugin`
2. `public diagnosticProvider: DiagnosticPlugin | undefined`
But I want to assign a default value to this property, so use method 2. 

However, users can only opt out by passing a Diagnostic class with empty track()